lib/omniauth/strategies/openid_connect.rb

```diff
@@ -109,7 +111,8 @@ def callback_phase
 
         raise CallbackError, 'Invalid state parameter' if invalid_state
 
-        return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(params['error'])) unless params['code']
+        grant_type_errors = validate_grant_type_credentials
+        return fail!(*grant_type_errors) if grant_type_errors.present?
```

Comment:
*Do not* use ActiveSupport's sugar like `#present?`. This gem can be used outside of rails projects.

Reply:
As I mentioned in my other comment, I'm not sure if you are aware but this gem requires active_support anyway and this is the reason why I used both `present?` and `constantize` methods.